internal/sql/sql.go

```diff
@@ -0,0 +1,190 @@
+package sql
+
+import (
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"math/rand"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+	"lukechampine.com/frand"
+)
+
+const (
+	maxRetryAttempts = 30  // 30 attempts
+	factor           = 1.8 // factor ^ retryAttempts = backoff time in milliseconds
```

Comment:
So on attempt 4 we're sleeping between 10 and 15 ms? I think the way we calculate the backoff is somewhat out of the ordinary, don't you usually have a base delay: `base * factor ** attempt`? Maybe a 50ms base combined with a factor of 1.8 makes sense for us. I'd probably even go for 1.6 and 1.7. 

```
Attempt #	Delay (ms)
1	50.0
2	90.0
3	162.0
4	291.6
5	524.88
6	944.78
7	1700.61
8	3061.10
9	5509.98
10	9917.96
```

Reply:
I think the basic exponential backoff is just `base ^ retries`. e.g. [link](https://en.wikipedia.org/wiki/Exponential_backoff).

The factor here is honestly just copied since it seems to work reasonably well for `hostd` and I've always felt like our current steps tend to sleep too much.